dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1199 @@
+# -*- coding: utf-8 -*-
+"""Creates a GCP cloud resource tree."""
+
+import json
+import tempfile
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+from typing import Dict, List, Optional, Any, Set
+from enum import Enum, auto
+
+from dateutil import parser
+from libcloudforensics.providers.gcp.internal.common import CreateService
+from libcloudforensics.providers.gcp.internal.log import GoogleCloudLog
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class OperatingMode(Enum):
+  """Enum represent operational mode (Online or Offline)."""
+
+  ONLINE = auto()
+  OFFLINE = auto()
+
+
+class Resource:
+  """A Class that represent a resource (Instance, Disk, Image...etc)."""
+
+  has_dynamic_attributes = True  # silences all attribute-errors for Resource
+
+  def __init__(self) -> None:
+    """Initialize the Resource object."""
+    self.id = ''
```

Comment:
missing type annotation for type str, please add type str here and to all the str arguments below

Reply:
Done